api/src/paths/artifact/intake.ts

```diff
@@ -0,0 +1,188 @@
+          }
+        }
+      }
+    }
+  },
+  responses: {
+    200: {
+      description: 'Submission OK.',
+      content: {
+        'application/json': {
+          schema: {
+            type: 'object',
+            
+          }
+        }
+      }
+    },
+    ...defaultErrorResponses
+  }
+};
+
+export function intakeArtifacts(): RequestHandler {
+  return async (req, res) => {
+    if (!req?.files?.length) {
+      throw new HTTP400('Missing required `media`');
+    }
+
+    const files: Express.Multer.File[] = Object.values(req.files);
+    if (req.body.metadata?.length !== files.length) {
+      throw new HTTP400('File count and metadata record count do not agree');
+    }
+
+    const dataPackageId = req.body?.data_package_id;
+    if (!dataPackageId) {
+      throw new HTTP400('Data package ID is required');
+    }
+
+    // Scan all files for viruses
+    const virusScanResults = await Promise.all(files.map(async (file) => scanFileForVirus(file)));
+    if (virusScanResults.some((result: boolean) => !result)) {
+      throw new HTTP400('Malicious content detected, upload cancelled');
+    }
+
+    const sourceSystem = getKeycloakSource(req['keycloak_token']);
+    if (!sourceSystem) {
+      throw new HTTP400('Failed to identify known submission source system', [
+        'token did not contain a clientId/azp or clientId/azp value is unknown'
+      ]);
+    }
+
+    const connection = getServiceAccountDBConnection(sourceSystem);
+
+    try {
+      await connection.open();
+
+      const submissionService = new SubmissionService(connection);
+      const artifactService = new ArtifactService(connection);
+
+      // Fetch the source transform record for this submission based on the source system user id
+      const sourceTransformRecord = await submissionService.getSourceTransformRecordBySystemUserId(
+        connection.systemUserId()
+      );
+
+      if (!sourceTransformRecord) {
+        throw new HTTP400('Failed to get source transform record for system user');
+      }
+
+      const metadataRecords: string[] = req.body.metadata;
+      const { source_transform_id } = sourceTransformRecord;
+
+      // Create a new submission for the artifact collection
+      const { submission_id } = await submissionService.getOrInsertSubmissionRecord({
+        source_transform_id: sourceTransformRecord.source_transform_id,
+        uuid: dataPackageId
+      });
+      
+      // Retrieve the ID that will be assigned to the artifact once it's inserted
+      const insertArtifactId = await artifactService.getNextArtifactId();
+
+      const artifactPersistPromises = files.map(async (file: Express.Multer.File, fileIndex: number) => {
+        const artifactMetadata: IArtifactMetadata = JSON.parse(metadataRecords[fileIndex]);
+        const artifactId = insertArtifactId + fileIndex;
+
+        // Generate the S3 key for the artifact, using the preemptive artifact ID + the package UUID
+        const s3Key = generateS3FileKey({
```

Comment:
Still undecided on what the final S3 path will be, but right now it's `platform/<UUID>/artifacts/<artifactID>/<file>`

Reply:
I think we will need to change it so that the `artifactID` is the final portion of the path. 

Looking at the artifact table, I dont see a key column, so I think we will need the S3 path to be deterministic:
`platform/<UUID>/artifacts/<artifactID>`

Reply:
Not sure of the top of my head what advantage this provides over just having a key column, but that seems to be how it is set up currently